docs/tutorial/en/source/tutorial/rag.py

```diff
@@ -0,0 +1,294 @@
+# -*- coding: utf-8 -*-
+"""
+.. _rag:
+
+Retrieval Augmentation Generation (RAG)
+==================================================================
+
+Agentscope has built-in supports for the retrieval augmentation generation
+(RAG). There are two key moduls related to RAG in AgentScope: `Knowledge` and
+`KnowledgeBank`.
+
+Create and Use Knowledge Instances
+----------------------------------------------
+
+While `Knowledge` is a base class, two specific built-in knowledge classes are in
+the AgentScope new.
+
+
+- `LlamaIndexKnowledge`: Designed to work with one of the most popular RAG library `LlamaIndex <https://www.llamaindex.ai/>`_ as local knowledge, and supporting most of LlamaIndex functionality by configuration.
+
+-  `BingKnowledge`: Designed to work with `Bing search <https://www.bing.com/>`_ to provide online knowledge.
+
+Create a `LlamaIndexKnowledge` instance
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A quick start to create a `LlamaIndexKnowledge` instance is to use the `build_knowledge_instance` function.
+There are three parameters need to be passed to the function.
+
+- `knowledge_id`: a unique identifier for this knowledge instance
+
+- `data_dirs_and_types`: a dictionary whose keys are strings of directories of the data, and values are the file extensions of the data
+
+- `emb_model_name`: name of the configuration of a embedding model in AgentScope (need to be initialized in AgentScope beforehand)
+
+A simple example is as follows.
+"""
+import os
+import agentscope
+from agentscope.rag.llama_index_knowledge import LlamaIndexKnowledge
+
+agent_list = agentscope.init(
```

Comment:
Suggested to initialize the agent explicitly, so that users can understand how to create or initialize the agent.